components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/ClaimUtil.java

```diff
@@ -305,6 +316,56 @@ private static String getSpMappedRoleClaim(ServiceProvider serviceProvider,
         return OIDCClaimUtil.getServiceProviderMappedUserRoles(serviceProvider, rolesList, claimSeparator);
     }
 
+    private static List<String> filterSharedRoles(ServiceProvider serviceProvider,
+                                                  List<String> userRoles) throws FrameworkException {
+
+        List<String> rolesAssociatedWithApp = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(userRoles)) {
+            rolesAssociatedWithApp = new ArrayList<>(userRoles);
+            /*
+             Check whether the application is a fragment application and if so, add only the shared roles
+             to the user's role list.
+            */
+            if (serviceProvider.getSpProperties() != null && Arrays.stream(serviceProvider.getSpProperties()).anyMatch(
```

Comment:
Let's apply the same comment for here also. Avoiding the code right shifting will increase the readability.